### PR TITLE
feat(ica-update): ica sizes and trending variant

### LINKED ISSUE #907: ICA Size Large
### Description

The ICA / ICAModule needs a large size variation. An additional variable can be supplied to ICA in order to change the size from the default settings. The large version of the ICA uses $heading-06 for the value and $heading-03 for the total. Are there any other requirements / changes for the large version?

image

### Goals / Use cases

- [ ] Provide alternative way to display ICA

### PR DESCRIPTION
**Pull request by CassidyJensen**
## Pull request 
Added sizes default, lg, and xlg to the ICA component. Added trending variant.

### Affected issues

- Resolves #907 and #908

### Proposed changes

- Styling for large and xlarge ICA
- Added trending icon 
- Added large and xlarge for trending. 
- Improved styling to handle ICAModule and ICAWall cases with the trending arrow.

### Testing instructions

- Go to the ICA story on storybook
- Use the ICA size and trending knobs to test the different sizes. (Should match the styling below).
image

### CODE REVIEW

**Review by cameroncalder — CHANGES_REQUESTED** [review]
Hey Cassidy, great steps forward here! Have a few notes:


**Universal feedback**
When the “percentage knob” is selected, it seems to restrict the ICA number sizes to change

The percentage should be placed on the same optical baseline as the large number (I’ll include spacing specs. Some include combinations of spacer tokens since in some cases there isn’t one single spacer that works perfectly on its own)

“Percentage” font size should be equal to the “Total” font size

The “total” should be displayed to the right of the large number by default and only shown below the number if space is an issue (position



**XLarge version**
The label font should change to body-short-01

**Comment by cameroncalder** [comment]
An updated image that includes spacer placement for the "Percentage" and "Total"

Artboard

**Review by cameroncalder — COMMENTED** [review]
Approved with one minor comment. Would you mind removing the space between the "/" and 
 the total value

**Comment by cameroncalder** [comment]
This is looking great 👍

**Comment by CassidyJensen** [comment]
> Approved with one minor comment. Would you mind removing the space between the "/" and
> the total value

oops! I will remove that now.

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:195]: For consistency, could we align with [Carbon's naming for sizes](https://react.carbondesignsystem.com/?path=/docs/components-button--default#button-size)? For example, `lg` instead of `large`.

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:194]: Because the type is generated from the propType, we don't need to add the additional `@type` annotation here.

```suggestion
  /** The size of the ICA. */
```

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:119]: As I understand, with the inclusion of `ICAClasses`, this will output `security--ica__label security--ica security--ica--large__label`, which could cause unintended side effects.

If this is to hook into styles for sizes, would it be sufficient to use a selector like `&--lg &__label`? This could potentially be applied anywhere `ICAClasses` is used.

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:132]: Similarly to above, [`classnames`](https://www.npmjs.com/package/classnames#usage) can be leveraged to conditionally handle these classes.

File: src/components/ICA/_index.scss
Comment (SimonFinney) [inline src/components/ICA/_index.scss:30]: We have hit consumer Sass conflicts in the past for not namespacing dependency spacing variables - here's an example of how this can be applied:

```suggestion
    padding-top: $carbon--spacing-01 + $carbon--spacing-04;
```

**Review by SimonFinney — COMMENTED** [review]
Thanks for your patience, and your first contribution -the design looks fantastic and spot on! 🎉

I've added a few comments and suggestions for your consideration - additionally, have you considered the kind of unit tests that could be included with these new variants?

**Review by cameroncalder — APPROVED** [review]
LGTM!

File: src/components/ICA/ICA.js
Comment (CassidyJensen) [reply src/components/ICA/ICA.js:119]: I removed the redundancy in the ICAClasses setup. It should now output `security--ica__label  security--ica--large`. Does this seem like a good fix?

**Comment by CassidyJensen** [comment]
> Thanks for your patience, and your first contribution -the design looks fantastic and spot on! 🎉
> 
> I've added a few comments and suggestions for your consideration - additionally, have you considered the kind of unit tests that could be included with these new variants?

@SimonFinney 
Not yet. Writing unit tests is completely new for me. What would you recommend? 

I have a couple ideas based on the [Testing Documentation](https://github.com/carbon-design-system/ibm-security/tree/dev/docs/testing).
There should be a test for the ICAClasses because they are custom classes correct? 
Would there need to be any additional accessibility testing or would the existing tests still cover it?

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:186]:
```suggestion
  size: PropTypes.oneOf(['default', 'lg', 'xl']),
```

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:191]:
```suggestion
  /** Display trending icon. */
```

File: src/components/ICA/ICA.js
Comment (SimonFinney) [reply src/components/ICA/ICA.js]: Apologies for any confusion so far - we can keep the `icaClasses` on the return function's root node, and remove it everywhere else as the styles can handle it from the top level. For example:

```js
  return (
    <div className={icaClasses} {...other}>
      <h4 className={`${namespace}__label`}>{label} </h4>
      ...
```

and in the styles:

```scss
@include security--component($name: ica) {

  &--lg & {
    &__label {
      ...
    }

    &__value {
      ...
    }
  }

  &--xl & {
    &__label {
      ...  
    }
  }

  ...
}
```

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:81]: For naming consistency with [Carbon's sizes](https://react.carbondesignsystem.com/?path=/docs/components-button--default#button-size):

```suggestion
  const isLarge = isSize('lg');
  const isXLarge = isSize('xl');
```

File: src/components/ICA/ICA.js
Comment (SimonFinney) [inline src/components/ICA/ICA.js:92]: For naming consistency with [Carbon's sizes](https://react.carbondesignsystem.com/?path=/docs/components-button--default#button-size):

```suggestion
    [`${namespace}--xl`]: isXLarge,
```

**Comment by SimonFinney** [comment]
Thanks @CassidyJensen ! As mentioned on Slack, if it helps you could submit the unit tests in a follow-up. Let me know if you haven't any questions on the final few pieces of feedback. 🔥

**Comment by CassidyJensen** [comment]
Doing a follow up with the unit tests would be great! Thank you. 
@cameroncalder Once I get the final approval from @SimonFinney this should be ready to wrap up!

**Review by SimonFinney — APPROVED** [review]
LGTM! 🔥 Thank you!

**Pull request merged**
